packages/@uppy/aws-s3-multipart/src/MultipartUploader.js

```diff
@@ -202,6 +271,13 @@ class MultipartUploader {
 
   _uploadPartBytes (index, url, headers) {
     const body = this.chunks[index]
+    const { signal } = this.abortController
+
+    let defer
+    const promise = new Promise((resolve, reject) => {
+      defer = { resolve, reject }
```

Comment:
so one downside I see to this approach is that if any unhandled error occurs anywhere in this function, it will not be caught by the Promise, and it will remain unhandled, and possibly unseen. But if we instead house the entire function inside the body of the promise:

```js
return new Promise((resolve, reject) => {
  // entire function body here
  ...
})
```
It would always catch any unforeseen errors that may be thrown.

Reply:
True, but if an error is thrown inside the synchronous code path here, it's a bug. So I don't think we need to try to catch it really 🤔